src/screens/LegacyWalletConnect/WalletConnectConnectorRequest.js

```diff
@@ -19,69 +19,51 @@
 */
 
 import * as React from 'react';
-import styled, { withTheme } from 'styled-components/native';
 import { Keyboard } from 'react-native';
-import t from 'translations/translate';
 import { useNavigation } from 'react-navigation-hooks';
+import styled from 'styled-components/native';
+import t from 'translations/translate';
 
-// components
-import ContainerWithHeader from 'components/Layout/ContainerWithHeader';
-import Image from 'components/Image';
+// Components
 import { Footer, ScrollWrapper } from 'components/Layout';
 import { Label, MediumText } from 'components/Typography';
-import Title from 'components/Title';
 import Button from 'components/Button';
+import ContainerWithHeader from 'components/Layout/ContainerWithHeader';
+import Image from 'components/Image';
+import Title from 'components/Title';
 
-// utils
-import { spacing, fontSizes } from 'utils/variables';
-import { images } from 'utils/images';
-
-// hooks
+// Hooks
 import useWalletConnect from 'hooks/useWalletConnect';
 
-// types
-import type { Theme } from 'models/Theme';
-
-
-type Props = {
-  theme: Theme,
-};
-
-
-const FooterWrapper = styled.View`
-  flex-direction: column;
-  width: 100%;
-`;
-
-const LabeledRow = styled.View`
-  margin: 10px 0;
-`;
-
-const Value = styled(MediumText)`
-  font-size: ${fontSizes.big}px;
-`;
-
-const OptionButton = styled(Button)`
-  margin-top: 4px;
-  flex-grow: 1;
-`;
+// Utils
+import { chainFromChainId, mapChainToChainId } from 'utils/chains';
+import { useThemedImages } from 'utils/images';
+import { useChainsConfig } from 'utils/uiConfig';
+import { spacing, fontSizes } from 'utils/variables';
+import { parsePeerName, pickPeerIcon } from 'utils/walletConnect';
 
+// Types
+import { CHAIN } from 'models/Chain';
 
-const WalletConnectConnectorRequestScreen = ({ theme }: Props) => {
+const WalletConnectConnectorRequestScreen = () => {
   const navigation = useNavigation();
+  const { genericToken } = useThemedImages();
+
   const { approveConnectorRequest, rejectConnectorRequest } = useWalletConnect();
 
-  const {
-    description,
-    url,
-    icons,
-    name,
-  } = navigation.getParam('peerMeta', {});
+  const peerMeta = navigation.getParam('peerMeta', {});
   const peerId = navigation.getParam('peerId');
 
+  const initialChainId = navigation.getParam('chainId');
+  const chain = chainFromChainId[initialChainId] ?? CHAIN.ETHEREUM;
+
+  // Note: this will map chain id to testnet in test env.
+  const chainId = mapChainToChainId(chain);
+  const chainConfig = useChainsConfig()[chain];
```

Comment:
A lot of `chain`'s here - could `chain` be `chainName` for naming consistency?

Reply:
Whilst it might be an `enum`, ultimately the variable references the name of the chain. It's not a showstopper, but just for readability.

Reply:
In other places value of this type are referred as `chain` and used as object key, e.g. chain balances, chain UI config, etc. IMO renaming to `chainName` could suggest that this is a user visible name of the chain. wdyt?